runLocally.sh

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Red Hat, Inc. and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+function confirm() {
+  declare -l answer # -l converts the value to lower case before it's assigned
+  read -r -p "$1 [y/N]: " "answer"
+  [[ "$answer" == "y" ]]
+}
+
+function abort() {
+  echo "Aborted."
+  exit 0
+}
+
+function standalone_jar_or_maven() {
+  echo
+  echo "Getting project version..."
+  # TODO fix mvnw's stdout handling
+  # TODO check if mvn is executable
+  readonly version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+  echo "Project version: ${version}"
+
+  readonly standalone=optaweb-vehicle-routing-standalone
+  readonly jar=${standalone}/target/${standalone}-${version}.jar
+
+  if [[ ! -f ${jar} ]]
+  then
+    confirm "Jarfile ‘$jar’ does not exist. Run Maven build now?" || abort
+    if ! ./mvnw clean install -DskipTests
+    then
+      echo >&2 "Maven build failed. Aborting the script."
+      exit 1
+    fi
+  fi
+}
+
+function validate() {
+  local osm_file_path=${osm_dir}/${osm_file}
+  local gh_graph_path=${gh_dir}/${osm_file%.osm.pbf}
+  if [[ ! -f "$osm_file_path" && ! -d "$gh_graph_path" ]]
+  then
+    echo >&2 "Wrong region ‘$osm_file’. One of the following must exist:"
+    echo >&2 "- OSM file: $osm_file_path"
+    echo >&2 "- GraphHopper graph: $gh_graph_path"
+    exit 1
+  fi
+}
+
+function run_optaweb() {
+  java -jar "${standalone}/target/${standalone}-${version}.jar" \
+ "--app.persistence.h2-dir=$vrp_dir/db" \
+ "--app.routing.osm-dir=$osm_dir" \
+ "--app.routing.gh-dir=$gh_dir" \
+ "--app.routing.osm-file=$osm_file" \
+ "--app.region.country-codes=$cc_list"
+}
+
+function download() {
+  local osm_url="http://download.geofabrik.de/$1"
+  echo "Downloading $osm_url..."
+  curl "$osm_url" -o "$2"
+  echo
+  echo "Created $2."
+}
+
+function country_code() {
+  local region=${1%.osm.pbf}
+  local cc_file=${cc_dir}/${region}
+
+  if [[ ! -f ${cc_file} ]]
+  then
+    region=${region%-latest}
+    region=${region//-/ }
+
+    local cc_tag="nv-i18n-1.27"
+    local cc_java="$vrp_dir/CountryCode-$cc_tag.java"
+    [[ ! -f ${cc_java} ]] && \
+      curl https://raw.githubusercontent.com/TakahikoKawasaki/nv-i18n/${cc_tag}/src/main/java/com/neovisionaries/i18n/CountryCode.java -s > "$cc_java"
```

Comment:
What does this do?

Reply:
This downloads a Java enum that is used to guess a country code from the OSM file name.